pkg/storage/memstore/memstore.go

```diff
@@ -130,6 +133,26 @@ func (m *MemStore) Get(k string) ([]byte, error) {
 	return v, nil
 }
 
+// GetBulk fetches the values associated with the given keys. This method works in an all-or-nothing manner.
+// It returns an error if any of the keys don't exist. If even one key is missing, then no values are returned.
+func (m *MemStore) GetBulk(keys ...string) ([][]byte, error) {
+	storedValues := make([][]byte, len(keys))
+
+	m.mux.RLock()
+	defer m.mux.RUnlock()
+
+	for i, key := range keys {
+		v, exists := m.db[key]
+		if !exists {
+			return nil, fmt.Errorf(getBulkKeyNotFound, key, storage.ErrValueNotFound)
```

Comment:
Same as comment for CouchDB store: Instead of returning an error, I would suggest that you set this entry to nil. The caller will realize that one of the documents he's requested is nil and therefore deleted or not found.

Reply:
Just talked with @bstasyszyn - the use case for this method is within some internal document processing in the EDV server, and in that case if any retrievals fail then the whole thing has to be considered a failure.